my-app/src/pages/LoginPage.tsx

```diff
@@ -0,0 +1,106 @@
+import * as React from 'react';
+import './LoginPage.css';
+import { render } from '@testing-library/react';
+import { FormErrors } from "./FormsErrors";
+
+interface LoginStates {
+    email: string;
+    password: string;
+    formErrors: {email:string, password: string};
+    emailValid: boolean,
+    passwordValid: boolean,
+    formValid: boolean,
+    ClickValid: boolean,
+}
+
+class LoginPage extends React.Component<{},LoginStates> {
+
+    constructor (props) {
+        super(props);
+        this.state = {
+          email: '',
+          password: '',
+          formErrors: {email: '', password: ''},
+          emailValid: false,
+          passwordValid: false,
+          formValid: false,
+          ClickValid: false,
```

Comment:
Acho que existem estados aqui que são redundantes, ou seja, você conseguiria inferir alguns desses valores a partir de outros. Exemplo: `formValid` pode ser inferido por uma lógica booleana simples entre `passwordValid` e `emailValid`, certo?

Como vc acha que pode melhorar a construção do seu estado?

Reply:
No momento eu acabo não usando o formValid, mas eu achei que seria necessário para futuramente passar de página. Mas acredito que da pra utilizar apenas um AND mesmo.